src/font/face.zig

```diff
@@ -222,6 +222,16 @@ pub const RenderOptions = struct {
             y: f64,
         };
 
+        /// Returns true if the constraint does anything. If it doesn't,
+        /// because it neither sizes nor positions the glyph, then this
+        /// returns false.
+        pub inline fn doesAnything(self: Constraint) bool {
```

Comment:
An idea... `Size` and `Align` can both fit into a `u2` (separately), so all 4 of these fields could fit into a `packed struct` as a `u8`, then you could do a `!= 0` comparison here.

It would probably save space on our structure too, not sure how much its used.

Reply:
Yeah, I considered making the constraint a packed struct as a whole, but it has a nullable field in it. I guess just condensing those specific fields would work though (and would make a smaller comparison than comparing the whole struct to something). I wasn't super concerned about performance since rasterizing glyphs isn't high throughput since it's all cached, but I wouldn't object to this change.